visualization-project-laptop-sales/visualization_project_laptop_sales_sol.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+"""visualization-project-laptop-sales_sol
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/10M53HHyac1sf2zjJ_UkB59movx5UKI0y
+"""
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import datetime
+
+"""# **#1 What is target market? Male or female?**"""
+
+sales = pd.read_csv(r'/content/Sales_data.txt')
+sales
+
+# Comparing 'Profit' between male and female
+total_profit = sales.groupby('Contact Sex')['Profit'].sum()
+total_profit
+
+"""1. The target marget seems to be males since their total profit of 2936.53 is higher than of females, which is 2522.48.
+
+# **#2 If business is cash-constrained, which gender should be targeted?**
+"""
+
+# Let's look at the 'Our Cost' column
+
+business_cost = sales.groupby('Contact Sex')['Our Cost'].sum()
+business_cost
+
+"""2. If the business is cash-constrained, they should probably target females more.
+
+# **#3 If consumer is cash-constrained, which gender should be targeted?**
+"""
+
+# Let's look at the average price of 'Sale Price' grouped by gender
+avg_consumer_cost = sales.groupby('Contact Sex')['Sale Price'].mean()
+avg_consumer_cost
+
+"""3. If the consumer is cash-constrained, males should be targeted.
+
+# **#4 What is our target age to maximize profit?**
+"""
+
+# First consider the individual ages
+
+individual_ages = sales.groupby('Contact Age')['Contact Age'].value_counts()
+individual_ages
+
+# Consider age ranges now
+age_ranges_5 = sales.groupby('Contact Age')['Contact Age'].hist(bins = list(range(20, 61, 5))) # Increments of 5
```

Comment:
You want to look at the sum of profits for each age range (hint: first use pd.cut() to make age bins, then group-sum profits by age and then again group-sum that dataframe on the age bins).